File: backend/model/control.go
package model

import "github.com/google/uuid"

type Control struct {
	Model
	// attribute
	Name        string `json:"name"`
	Key         string `json:"key"`
	Description string `json:"description"`
	// fk
	BoxId    uuid.UUID     `json:"boxId"`
	LayoutId uint          `json:"layoutId"`
	WidgetId uint          `json:"widgetId"`
	Widget   WidgetControl `json:"widget"`
	// relation
	ControlData []ControlData `gorm:"foreignKey:ControlId" json:"controlData"`
	Layout      Layout        `json:"layout"`
}

File: web/src/types/Control.ts
import { DataControl } from './Data';
import { Layout } from './Layout';
import { Widget } from './Widget';

export interface Control {
  id: number;
  createdAt: Date;
  updatedAt: Date;
  name: string;
  key: string;
  description: string;
  BoxId: string;
  layoutId: number;
  widgetId: number;
  controlData: DataControl[];
  widget: Widget;
  layout: Layout;
}
